src/trpc/router/restRouter.ts

```diff
@@ -625,6 +627,72 @@ export const restRouter = router({
         )
         .query(({ input }) => achievementMethods.getList(input)),
 
+    giveAchievementForHireSections: restProcedure
+        .meta({
+            openapi: {
+                method: 'POST',
+                path: '/achievements/sections',
+                protect: true,
+                summary: 'Give user crew achievement for hire sections',
+            },
+        })
+        .input(
+            giveAchievementSchema
+                .omit({ userId: true, achievementTitle: true, achievementId: true, amount: true })
+                .extend({
+                    targetUserEmail: z.string(),
+                    actingUserEmail: z.string(),
+                    sectionsNumber: z.number(),
+                }),
+        )
+        .output(z.string())
+        .query(async ({ input, ctx }) => {
+            const { actingUserEmail, targetUserEmail, sectionsNumber } = input;
+
+            if (!config.sectionAchiementId || !config.sectionAmountForAchievement) {
+                throw new TRPCError({
+                    code: 'NOT_FOUND',
+                    message: 'No sectionAchiementId or sectionAmountForAchievement in crew config',
+                });
+            }
+
+            const [targetUser, actingUser, achievement] = await Promise.all([
+                userMethods.getUserByField({ email: targetUserEmail }),
+                userMethods.getUserByField({ email: actingUserEmail }),
+                achievementMethods.getById(config.sectionAchiementId),
+            ]);
+
+            if (sectionsNumber % Number(config.sectionAmountForAchievement) !== 0) {
+                return 'Amount of completed sections not divided by sectionAmountForAchievement';
```

Comment:
Amount of completed sections is not evenly divisible by sectionAmountForAchievement